source/simulator/helper_functions.cc

```diff
@@ -909,8 +909,142 @@ namespace aspect
 
     return (material_model->get_model_dependence().viscosity != MaterialModel::NonlinearDependence::none);
   }
-}
 
+
+
+// apply limiter for dg solutions
+  /*
+   * normalize the pressure by calculating the surface integral of the pressure on the outer
+   * shell and subtracting this from all pressure nodes.
+   */
+  template <int dim>
+  void Simulator<dim>::apply_limiter_to_dg_solutions(LinearAlgebra::BlockVector &vector)
+  {
+    // create a quadrature formula based on the compositional element alone.
+    // be defensive about determining that a compositional field actually exists
+    if ( parameters.n_compositional_fields== 0)
+      AssertThrow (introspection.base_elements.compositional_fields
+                   != numbers::invalid_unsigned_int,
+                   ExcMessage("This postprocessor cannot be used without compositional fields."));
+
+    const QGauss<1> quadrature_formula_x(parameters.composition_degree+1);
```

Comment:
It would be useful to either choose more explanatory descriptions than x and y, or to add a comment above this line to explain what x and y means.